File: 4229-trim-trailing-vowels/trim-trailing-vowels.cpp
class Solution {
public:
    bool isVowel(char &c) {
        if(c == 'a' || c == 'i' || c == 'o' || c == 'u' || c == 'e') {
            return true;
        }
        return false;
    }
    string trimTrailingVowels(string s) {
        while(s.length() > 0) {
            if(isVowel(s.back())) {
                s.pop_back();
            }
            else return s;
        }
        return s;
    }
};

File: 4229-trim-trailing-vowels/trim-trailing-vowels.java
class Solution {
    private boolean isVowel (char c) {
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') return true;
        return false;
    }
    public String trimTrailingVowels(String s) {
        StringBuilder sb = new StringBuilder(s);

        while(sb.length() > 0) {
            if(isVowel(sb.charAt(sb.length() -1))) {
                sb.deleteCharAt(sb.length() - 1);
            } else {
                return sb.toString();
            }
        }

        return sb.toString();
    }
}